<a href="https://colab.research.google.com/github/minsaee/ai_chat_project01/blob/master/003_goorm_settings.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!python -V

Python 3.10.12


공개키가 오래 되어서 생기는 오류 -> 다음 서명이 올바르지 않습니다: EXPKEYSIG 172B5989FCD21EF8 CF CLI Team <cf-cli-eng@pivotal.io>

wget -q -O - https://packages.cloudfoundry.org/debian/cli.cloudfoundry.org.key | apt-key add -

실행후 sudo apt-get update

다음 서명들은 공개키가 없기 때문에 인증할 수 없습니다: NO_PUBKEY number

$ sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 536F8F1DE80F6A35

$ sudo apt-get update

구름 파이썬 업그레이드 순서  
터미널 입력  
 - sudo apt update
 - sudo apt install python 3.10

명령어 결과 나오는거 이용하기.

- which python3
- which python3.10
  
- sudo update-alternatives --install /usr/local/bin/python3 python /usr/bin/python3.10 1
/usr/local/bin/python3 에 결과1를 넣어주세요  
/usr/bin/python3.10에 결과2를 넣어주세요  

터미널 설치
 - pip install --upgrade pip  
 - pip install fastapi
 - pip install 'uvicorn[standard]'
 - pip install openai==0.28.1
 - pip install langchain
 - pip install chromadb  

 - pip install --upgrade pip==20.3

쉘스크립트 파일 하나 만들고 하고싶은이름.sh 이렇게 그리고 아래 코드 복붙



#!/bin/bash

sudo apt-get update

echo | sudo add-apt-repository ppa:deadsnakes/ppa

sudo apt install python3.10 --yes

sudo update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1

sudo update-alternatives --install /usr/bin/python python /usr/bin/python3.10 2

echo "1" | sudo update-alternatives --config python



터미널에서 sh 파일이름.sh 입력해서 실행 끝